pkg/azureclients/v2/client-gen/generator/template.go

```diff
@@ -55,7 +55,10 @@ func (client *Client) CreateOrUpdate(ctx context.Context, resourceGroupName stri
 	if err != nil {
 		return nil, err
 	}
-	return &resp.{{.Resource}}, nil
+	if resp != nil {
+		return &resp.{{.Resource}}, nil
+	}
+	return nil, nil
```

Comment:
does `nil, nil` mean resource not found? As a best practice, it's better returning a NotFound error on 404

Reply:
the client will return error when status code is 404